### DISCUSSION TITLE
Chapter 3 Linear and Non Linear FMNIST

### DISCUSSION THREAD

**Discussion by kiki101robo** [mrdbourke/pytorch-deep-learning#1058] [Q&A] [answered]
As instructed in the course, I first used linear model and then non linear model and compared their results. But my non linear layer seems to be working better or equivalent to my linear layer in contrast to what was demonstrated in the video follow along. Is it okay or am I doing something wrong?

![Non Linear](https://github.com/user-attachments/assets/617d4eb3-9930-490d-887e-99af29e3872e)
![image](https://github.com/user-attachments/assets/4232954b-4b6a-4290-b0b8-a8fed8e57a47)

**Comment by bmaxdk** [comment, marked as answer]
It's perfectly fine if your nonlinear model performs better or at least equivalent to the linear model. The fact is this is expected in many cases since nonlinear models can capture more complex patterns in the data compared to linear models, which are restricted to linear relationships.

For the result you see, where the nonlinear model has higher accuracy and lower loss, suggest that the nonlinear model is better able to fit the data. However, the performance difference between linear and nonlinear models can vary depending on factors like the dataset, model architecture, and hyperparameters.

Nonlinear models generally have more flexibility to learn complex patterns, which can lead to better performance.However, be careful with overfitting. If the nonlinear model is significantly more complex, it might fit the training data very well but not generalize as well to unseen data.

Good luck with your studying!

**Reply by kiki101robo** [reply]
Thanks for the explanation!
